File: uav_control/src/simple_gripper.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include <sstream>

int main(int argc, char **argv){

ros::init(argc, argv, "gripper");
ros::NodeHandle n;

ros::Rate loop_rate(10);

ros::Subscriber gripper_sub = n.subscribe("grip_command", 1000);

//ros::Publisher gripper_pub = n.advertise<geometry_msgs::Twist>("gripper", 100);

//geometry_msgs::Twist msg;

if (gripper_sub == "open"){

msg.data = 0.8;

gripper_pub.publish(msg);
}

if (gripper_sub == "close"){

msg.data = 0.2;

gripper_pub.publish(msg);
}

ros::spin();

return 0;
}

File: uav_control/src/simple_gripper.py
#!/usr/bin/env python
# license removed for brevity
import rospy
from std_msgs.msg import String

def gripper():

    rospy.wait_for_service('AttachObject')
    rospy.init_node('gripper', anonymous=True)
    rospy.Subscriber("grip_attach", String, callback)
    att = rospy.ServiceProxy("AttachObject",AttachObject)
    rate = rospy.Rate(10) # 10hz
    rate.sleep()

    if (data.data == "Close")
	resp = att(True, box_name)
    if (data.data == "open")
	execAction = att(False, box_name)

if __name__ == '__main__':
    try:
	gripper()
    except rospy.ROSInterruptException:
        pass
